### PR TITLE
WIP: Add base types

### PR DESCRIPTION
**Pull request by Sergyus**

### CODE REVIEW

File: metacom.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  constructor(url: string)
+  /**
+   * ready
+   */
+  ready(): Promise
+  /**
+   * load folder
+   * @param group
+   */
+  load(...group: string): Promise<void>
```
Comment (Sergyus) [inline metacom.d.ts:28]:
```suggestion
  load(...group: string[]): Promise<void>

```

File: metacom.d.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  socket: WebSocket
+  /** api */
+  api: any
+  /** callId */
+  callId: number
+  /** calls */
+  calls: Map<any, any>
+  /**
+   * constructor
+   * @param url host
```
Comment (Sergyus) [inline metacom.d.ts:17]:
```suggestion
   * @param url host
   * @example 'wss://127.0.0.1:8000'
```

Comment (tshemsedinov) [reply metacom.d.ts:17]: Do we really need jsdoc in d.ts?

File: metacom.d.ts (hunk 1 above)
Comment (tshemsedinov) [reply metacom.d.ts:28]: You can add fixup to this PR, to do this just add this changes locally and run following:
```
git add -A
git commit --fixup=bb2dfe4db75b533617c9b9f3468dc89b22fa6e2f
```

File: package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "url": "https://github.com/metarhia/metacom"
   },
   "main": "metacom.js",
+  "types": "./metacom.d.ts",
```
Comment (tshemsedinov) [inline package.json:27]: Try to remove `./`

**Comment by tshemsedinov** [comment]
Closed in favour of https://github.com/metarhia/metacom/pull/162

**Pull request closed without merging**